Tests/SwiftLintFrameworkTests/RequiredEnumCaseRuleTestCase.swift

```diff
@@ -24,7 +24,7 @@ class RequiredEnumCaseRuleTestCase: XCTestCase {
 
     func testRequiredCaseHashValue() {
         let requiredCase = RequiredCase(name: "success")
-        XCTAssertEqual(requiredCase.hashValue, "success".hashValue)
+        XCTAssertEqual(requiredCase.hashValue, RequiredCase(name: "success").hashValue)
```

Comment:
What's the point in testing `XCTAssertEqual(RequiredCase(name: "success").hashValue, RequiredCase(name: "success").hashValue)`?

Reply:
The ideia is to make sure that `hashCode` is the same for enums containing associated values. 

This is now basically testing the compiler, but the fact that we're generating `hashValue` automatically is an implementation detail and I don't think it hurts to test it.

Note that we can't compare to a value directly because now the `hashValue` can (and will) return different values in different program runs.

Reply:
I'm not convinced these tests give much value, but fine to keep if you think they do.

Tests being `testRequiredCaseHashValue`, `testRequiredCaseEquatableReturnsTrue` and `testRequiredCaseEquatableReturnsFalseBecauseOfDifferentName`.